addons/hr_timesheet/controllers/portal.py

```diff
@@ -68,7 +75,16 @@ def portal_my_timesheets(self, page=1, sortby=None, filterby=None, search=None,
         domain = AND([domain, searchbar_filters[filterby]['domain']])
 
         if search and search_in:
-            domain = AND([domain, [('name', 'ilike', search)]])
+            search_domain = []
+            if search_in in ('project', 'all'):
+                search_domain = OR([search_domain, [('project_id', 'ilike', search)]])
```

Comment:
I think it should be `AND`? otherwise if the `search_domain` explicitely restrincts access to some records it is bypassed. (and the search is done with a sudo'ed env! => no access rights check)

Reply:
Just check again. I think here should be `OR`. Otherwise `search in All` will not work. And I think search here is not done with a sudo'ed env. `OR` and `AND` is only affecting `search in All`